### PR TITLE
add option for input validation foreground color

### LINKED ISSUE #57536: Add inputValidation.foreground as a theme token
In some instances, the `foreground` color does not work well with the `inputValidation` backgrounds. Here's an example where a theme has set `foreground` to `#888`:

image

This forces the theme author to choose only lighter colors that work with `#888`.

If we could set `inputValidation.foreground` to `#FFF` then that would work best with those backgrounds:

image

### PR DESCRIPTION
**Pull request by ParkourKarthik**
This closes #57536 

@bpasero I've made changes only related to "find Input control".
Review and let me know if similar changes could be applied to all other input validations.

### CODE REVIEW

File: src/vs/platform/theme/common/colorRegistry.ts [hunk 1]
```diff
@@ -180,10 +180,13 @@ export const inputActiveOptionBorder = registerColor('inputOption.activeBorder',
...
+export const inputValidationWarningForeground = registerColor('inputValidation.warningForeground', { dark: '#FFFFFF', light: '#063B49', hc: Color.black }, nls.localize('inputValidationWarningForeground', "Input validation background color for information severity."));
 export const inputValidationWarningBorder = registerColor('inputValidation.warningBorder', { dark: '#B89500', light: '#B89500', hc: contrastBorder }, nls.localize('inputValidationWarningBorder', "Input validation border color for warning severity."));
 export const inputValidationErrorBackground = registerColor('inputValidation.errorBackground', { dark: '#5A1D1D', light: '#F2DEDE', hc: Color.black }, nls.localize('inputValidationErrorBackground', "Input validation background color for error severity."));
+export const inputValidationErrorForeground = registerColor('inputValidation.errorForeground', { dark: '#FFFFFF', light: '#063B49', hc: Color.black }, nls.localize('inputValidationErrorForeground', "Input validation background color for information severity."));
```
Comment (bpasero) [inline src/vs/platform/theme/common/colorRegistry.ts]: This should be "foreground"

File: src/vs/platform/theme/common/colorRegistry.ts [hunk 2]
```diff
@@ -180,10 +180,13 @@ export const inputActiveOptionBorder = registerColor('inputOption.activeBorder',
...
+export const inputValidationInfoForeground = registerColor('inputValidation.infoForeground', { dark: '#FFFFFF', light: '#063B49', hc: Color.black }, nls.localize('inputValidationInfoForeground', "Input validation background color for information severity."));
 export const inputValidationInfoBorder = registerColor('inputValidation.infoBorder', { dark: '#007acc', light: '#007acc', hc: contrastBorder }, nls.localize('inputValidationInfoBorder', "Input validation border color for information severity."));
 export const inputValidationWarningBackground = registerColor('inputValidation.warningBackground', { dark: '#352A05', light: '#F6F5D2', hc: Color.black }, nls.localize('inputValidationWarningBackground', "Input validation background color for warning severity."));
+export const inputValidationWarningForeground = registerColor('inputValidation.warningForeground', { dark: '#FFFFFF', light: '#063B49', hc: Color.black }, nls.localize('inputValidationWarningForeground', "Input validation background color for information severity."));
```
Comment (bpasero) [inline src/vs/platform/theme/common/colorRegistry.ts]: This should be "foreground"

File: src/vs/platform/theme/common/colorRegistry.ts [hunk 3]
```diff
@@ -180,10 +180,13 @@ export const inputActiveOptionBorder = registerColor('inputOption.activeBorder',
 export const inputPlaceholderForeground = registerColor('input.placeholderForeground', { light: transparent(foreground, 0.5), dark: transparent(foreground, 0.5), hc: transparent(foreground, 0.7) }, nls.localize('inputPlaceholderForeground', "Input box foreground color for placeholder text."));
 
 export const inputValidationInfoBackground = registerColor('inputValidation.infoBackground', { dark: '#063B49', light: '#D6ECF2', hc: Color.black }, nls.localize('inputValidationInfoBackground', "Input validation background color for information severity."));
+export const inputValidationInfoForeground = registerColor('inputValidation.infoForeground', { dark: '#FFFFFF', light: '#063B49', hc: Color.black }, nls.localize('inputValidationInfoForeground', "Input validation background color for information severity."));
```
Comment (bpasero) [inline src/vs/platform/theme/common/colorRegistry.ts]: This should be "foreground"

**Review by bpasero — CHANGES_REQUESTED** [review]
@ParkourKarthik unless I am mistaken, these colors should not have any default set, otherwise we change the semantics, aren't we? Today I do not think we apply any particular foreground color by default and I think we should only do so if the theme defines this color explicitly.

Also, we need to support this color everywhere where the correlating background color is passed through, so please find all occurrences of where input background colors are used and pass them through.

**Comment by ParkourKarthik** [comment]
@bpasero I've set the default values to `null`. I believe that is meant to be done when no actual defaults are to be set.

Note: I've not added any Validation Foreground colors to the below as there should not be defaults for Foreground. I believe that it is expected.
https://github.com/Microsoft/vscode/blob/8539f48172f1ba5acfe34bb5633eb684d59634ba/src/vs/base/browser/ui/inputbox/inputBox.ts#L72-L81

**Review by bpasero — CHANGES_REQUESTED** [review]
@ParkourKarthik much better. There is one remaining usage of `inputValidationInfoBackground` in [`messageController`](https://github.com/Microsoft/vscode/blob/8539f48172f1ba5acfe34bb5633eb684d59634ba/src/vs/editor/contrib/message/messageController.ts#L193) that can also have a foreground color from the theme.

**Pull request merged**